fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJob.java

```diff
@@ -0,0 +1,409 @@
+            return;
+        }
+        // state can not be PENDING here
+        switch (info.getToState()) {
+            case OVERWRITE_PREPARED:
+                sourcePartitionNames = info.getSourcePartitionNames();
+                newPartitionNames = info.getNewPartitionsName();
+                jobState = OverwriteJobState.OVERWRITE_PREPARED;
+                break;
+            case OVERWRITE_FAILED:
+                jobState = OverwriteJobState.OVERWRITE_FAILED;
+                LOG.info("replay insert overwrite job:{} to FAILED", jobId);
+                gc();
+                break;
+            case OVERWRITE_CANCELLED:
+                jobState = OverwriteJobState.OVERWRITE_CANCELLED;
+                LOG.info("replay insert overwrite job:{} to CANCELLED", jobId);
+                gc();
+                break;
+            case OVERWRITE_SUCCESS:
+                jobState = OverwriteJobState.OVERWRITE_SUCCESS;
+                doCommit();
+                LOG.info("replay insert overwrite job:{} to SUCCESS", jobId);
+                break;
+            default:
+                LOG.warn("invalid to state:{} for insert overwrite job:{}", info.getToState(), jobId);
+                break;
+        }
+    }
+
+    private void transferTo(OverwriteJobState state) throws Exception {
+        if (state == OverwriteJobState.OVERWRITE_SUCCESS) {
+            Preconditions.checkState(jobState == OverwriteJobState.OVERWRITE_PREPARED);
+        }
+        InsertOverwriteStateChangeInfo info =
+                new InsertOverwriteStateChangeInfo(jobId, jobState, state,
+                        sourcePartitionNames, newPartitionNames);
+        GlobalStateMgr.getCurrentState().getEditLog().logInsertOverwriteStateChange(info);
+        jobState = state;
+        handle();
+    }
+
+    private void prepare() throws Exception {
+        Preconditions.checkState(jobState == OverwriteJobState.OVERWRITE_PENDING);
+        this.watershedTxnId =
+                GlobalStateMgr.getCurrentGlobalTransactionMgr().getTransactionIDGenerator().getNextTransactionId();
+        List<Partition> sourcePartitions = originalTargetPartitionIds.stream()
+                .map(id -> targetTable.getPartition(id)).collect(Collectors.toList());
+        sourcePartitionNames = sourcePartitions.stream().map(p -> p.getName()).collect(Collectors.toList());
+        newPartitionNames = sourcePartitionNames.stream().map(name -> name + postfix).collect(Collectors.toList());
+        transferTo(OverwriteJobState.OVERWRITE_PREPARED);
+    }
+
+    private void executeInsert() throws Exception {
+        long insertStartTimestamp = System.currentTimeMillis();
+        // should replan here because prepareInsert has changed the targetPartitionNames of insertStmt
+        ExecPlan newPlan = new StatementPlanner().plan(insertStmt, context);
+        stmtExecutor.handleDMLStmt(newPlan, insertStmt);
+        insertElapse = System.currentTimeMillis() - insertStartTimestamp;
```

Comment:
insertElapse is unused?

Reply:
it is used in log when job succeeds or fails